Add support for config file

cgeorgii opened the pull request:
Added information about the feature to the `README`, but in short it aims to work similarly to the `.rspec` configuration file for RSpec. Options set there can be overridden by passing arguments to the executable.

Added specs for it and tested with an app I maintain. Let me know if you have some comments or suggestions!

grodowski commented on lib/undercover/cli.rb:
```diff
@@ -33,6 +34,25 @@ def self.run(args)
     # rubocop:enable Metrics/MethodLength, Metrics/AbcSize
 
+    def self.build_opts(args)
+      configuration = project_options.concat(Array(args))
```
`CLI.run` is always called with an array, so I think it's ok to just use `args`

cgeorgii replied:
Oh, I ran into problems with this but it was just the way I was running the tests. Thanks, fixed.

grodowski commented on lib/undercover/version.rb:
```diff
@@ -1,5 +1,5 @@
 
 module Undercover
-  VERSION = '0.2.2'
+  VERSION = '0.3.2'
```
That could become just `0.3.0`, but I would suggest to revert this change for now. Could you add an entry to "unreleased" in `CHANGELOG.md` and I'll do the version bump with a release commit?

cgeorgii commented:
@grodowski Thanks for the comments! I moved the code to `Undercover::Options` and split `Undercover::CLI#run` in order to make it easier to test. Let me know if you have any other suggestions or thoughts. :)
